backends/candle/src/models/flash_jina_code.rs

```diff
@@ -0,0 +1,458 @@
+        let attention_head_size = config.hidden_size / config.num_attention_heads;
+        let all_head_size = config.num_attention_heads * attention_head_size;
+        let hidden_size = config.hidden_size;
+
+        let query_weight = vb
+            .pp("self.query")
+            .get((all_head_size, hidden_size), "weight")?;
+        let query_bias = vb.pp("self.query").get(all_head_size, "bias")?;
+        let key_weight = vb
+            .pp("self.key")
+            .get((all_head_size, hidden_size), "weight")?;
+        let key_bias = vb.pp("self.key").get(all_head_size, "bias")?;
+        let value_weight = vb
+            .pp("self.value")
+            .get((all_head_size, hidden_size), "weight")?;
+        let value_bias = vb.pp("self.value").get(all_head_size, "bias")?;
+
+        let layer_norm_q = LayerNorm::load(
+            vp.pp("self").pp("layer_norm_q"),
+            config.hidden_size,
+            config.layer_norm_eps as f32,
+        )?;
+        let layer_norm_k = LayerNorm::load(
+            vp.pp("self").pp("layer_norm_k"),
+            config.hidden_size,
+            config.layer_norm_eps as f32,
+        )?;
+
+        let query_linear = Linear::new(query_weight, Some(query_bias), None);
+        let key_linear = Linear::new(key_weight, Some(key_bias), None);
+        let value_linear = Linear::new(value_weight, Some(value_bias), None);
+
+        let dense_weight = vb
+            .pp("output")
+            .pp("dense")
+            .get((hidden_size, hidden_size), "weight")?;
+        let dense_bias = vb.pp("output").pp("dense").get(hidden_size, "bias")?;
+
+        let dense = Linear::new(dense_weight, Some(dense_bias), None);
+
+        let layer_norm_out = LayerNorm::load(
+            vb.pp("output").pp("LayerNorm"),
+            config.hidden_size,
+            config.layer_norm_eps as f32,
+        )?;
+
+        let softmax_scale = (1. / (attention_head_size as f64).sqrt()) as f32;
+
+        Ok(Self {
+            query_linear,
+            key_linear,
+            value_linear,
+            dense,
+            layer_norm_q,
+            layer_norm_k,
+            layer_norm_out,
+            alibi_slopes,
+            num_attention_heads: config.num_attention_heads,
+            attention_head_size,
+            softmax_scale,
+            span: tracing::span!(tracing::Level::TRACE, "attention"),
+        })
+    }
+
+    pub fn forward(
+        &self,
+        hidden_states: &Tensor,
+        cu_seqlens: &Tensor,
+        max_s: usize,
+    ) -> Result<Tensor> {
+        let _enter = self.span.enter();
+
+        let residual = hidden_states.clone();
+
+        let query_layer = self.query_linear.forward(hidden_states)?;
+        let query_layer = self.layer_norm_q.forward(&query_layer, None)?;
+
+        let key_layer = self.key_linear.forward(hidden_states)?;
+        let key_layer = self.layer_norm_k.forward(&key_layer, None)?;
+
+        let value_layer = self.value_linear.forward(hidden_states)?;
```

Comment:
You should still merge qkv and do one fused linear.
It's faster to do one big linear instead of 3 smaller ones.
You can then split the result and apply the layer norm for q and k.